helper_funcs.py

```diff
@@ -469,52 +486,270 @@ def _mask_flags(times, LSD):
+            file = containers.RFIMask.from_file(rfi_path)
+        except FileNotFoundError:
+            continue
+        rfm |= file.mask[:]
+
+    # Apply the initial weight mask
+    vis *= np.where(power.weight[:, 0] == 0, 1, np.nan)
+    # Apply the full mask
+    vis_mask = vis * np.where(rfm == 0, 1, np.nan)
+    vis_mask *= np.where(_mask_flags(power.time, LSD), np.nan, 1)
 
     cmap = copy.copy(matplotlib.cm.viridis)
-    cmap.set_bad("#aaaaaa")
+    cmap.set_bad(_BAD_VALUE_COLOR)
+
+    # Make the master figure
+    mfig = plt.figure(layout="constrained", figsize=(50, 20))
+    # MAke the two sub-figures
+    subfigs = mfig.subfigures(1, 2, wspace=0.1)
 
+    for fig, sim in zip(subfigs, (vis, vis_mask)):
+
+        axis = fig.subplots(1, 1)
+
+        im = axis.imshow(
+            sim,
+            extent=(0, 360, 400, 800),
+            cmap=cmap,
+            aspect="auto",
+            interpolation="nearest",
+            vmin=vmin,
+            vmax=vmax,
+        )
+        divider = make_axes_locatable(axis)
+        cax = divider.append_axes("right", size="1.5%", pad=0.25)
+        fig.colorbar(im, cax=cax)
+        axis.set_xlabel("RA [deg]", fontsize=30)
+        axis.set_ylabel("Freq [MHz]", fontsize=30)
+
+        # Calculate events like solar transit, rise ...
+        ev = events(chime_obs, LSD)
+        # Highlight the day time data
+        sr = (ev["sun_rise"] % 1) * 360 if "sun_rise" in ev else 0
+        ss = (ev["sun_set"] % 1) * 360 if "sun_set" in ev else 360
+
+        if sr < ss:
+            axis.axvspan(sr, ss, color="grey", alpha=0.4)
+        else:
+            axis.axvspan(0, ss, color="grey", alpha=0.4)
+            axis.axvspan(sr, 360, color="grey", alpha=0.4)
+
+        axis.axvline(sr, color="k", ls="--", lw=1)
+        axis.axvline(ss, color="k", ls="--", lw=1)
+
+        title = "rev 0" + str(rev) + ", LSD " + str(LSD)
+        axis.set_title(title, fontsize=50)
+        _ = axis.set_xticks(np.arange(0, 361, 45))
+
+
+@_fail_quietly
+def plotFactMask(rev, LSD):
+    path = _get_rev_path("fact_mask", rev, LSD)
+    fmask = containers.RFIMask.from_file(path)
+
+    mask = fmask.mask[:]
+
+    # Load all the RFI masks
+    for name in {"stokesi_mask", "sens_mask", "chisq_mask"}:
+        rfi_path = _get_rev_path(name, rev, LSD)
+        try:
+            file = containers.RFIMask.from_file(rfi_path)
+        except FileNotFoundError:
+            continue
+        try:
+            rfm |= file.mask[:]
+        except NameError:
+            # First mask to be loaded
+            rfm = file.mask[:].copy()
+
+    # Make the master figure
+    fig = plt.figure(layout="constrained", figsize=(15, 10))
+    axis = fig.subplots(1, 1)
+
+    # Plot the full mask
+    axis.imshow(
+        mask,
+        extent=(0, 360, 400, 800),
+        cmap="binary",
+        aspect="auto",
+        interpolation="nearest",
+    )
+    # Overlay the factorized mask
     im = axis.imshow(
-        sensrat,
+        rfm,
```

Comment:
This will raise `UnboundLocalError` if none of the RFI masks were found above.

In the other functions, you've initialised `rfm` to `np.zeroes` before trying to load the files.

Reply:
Yeah good catch. The reason I didn't initialize it the same way here is because the factorized mask is gridded to RA coordinates, whereas the RFI masks are all still time sampled, so they won't end up having the correct axes. I'll have to think a bit about how we want this to behave if none of the masks are loaded

Reply:
You could also just have some sort of `if "rfm" in locals():`' check to avoid the exception.